module/power_domain/src/mod_power_domain.c

```diff
@@ -1195,6 +1218,7 @@ static void process_power_state_transition_report(struct pd_ctx *pd,
         return;
     }
 
+#ifdef BUILD_HAS_NOTIFICATION
```

Comment:
I think, Below code is not related to notifications

if (pd->parent == NULL) {
/* this is the top pd (SYSTOP) /
if (mod_pd_ctx.system_suspend.state != MOD_PD_STATE_ON) {
/ has gone down, invalidate the system suspend ongoing */
mod_pd_ctx.system_suspend.suspend_ongoing = false;
}

Reply:
yes you're right